scripts/build/builders/tizen.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+from enum import Enum, auto
+
+from .gn import GnBuilder
+
+
+class TizenApp(Enum):
+    LIGHT = auto()
+
+    def ExampleName(self):
+        if self == TizenApp.LIGHT:
+            return 'lighting-app'
+        else:
+            raise Exception('Unknown app type: %r' % self)
+
+    def AppNamePrefix(self):
+        if self == TizenApp.LIGHT:
+            return 'chip-tizen-lighting-example'
+        else:
+            raise Exception('Unknown app type: %r' % self)
+
+    def FlashBundleName(self):
+        if self == TizenApp.LIGHT:
+            return 'lighting_app.flashbundle.txt'
+        else:
+            raise Exception('Unknown app type: %r' % self)
+
+
+class TizenBuilder(GnBuilder):
+
+    def __init__(self,
+                 root,
+                 runner,
+                 output_prefix: str,
+                 app: TizenApp = TizenApp.LIGHT):
+        super(TizenBuilder, self).__init__(
+            root=os.path.join(root, 'examples', app.ExampleName(), 'linux'),
+            runner=runner,
+            output_prefix=output_prefix)
+
+        self.app = app
+        tizen_home = os.environ['TIZEN_HOME']
+        os.environ['PKG_CONFIG_SYSROOT_DIR'] = tizen_home
```

Comment:
this changes the underlying OS environment.

Currently the builds scripts attempt to provide a `--dry-run` command that shows how to build things manually if needed and this is checked in unit tests. As a result, you should not be using python-code to run anything that actually touches the build. This includes changing environment or things like 'python file copies'.

Please pipe the 'environment changing' through to be visible in 'dry run'

Reply:
Thank you for comments! Done.